tracing/trace_manager.py

```diff
@@ -66,7 +66,7 @@ def _run_traces(self, trace_interval):
             if self.number_pending_requests() > 0:
                 try:
                     new_request_ids = []
-                    for req_id in self._request_queue:
+                    for req_id in self._request_queue.copy():
```

Comment:
Looks good. I've also spotted another shared data structure that's being iterated on and has insertions from different threads `self.trace_pkt_in` can get [appended from this handler](https://github.com/kytos-ng/sdntrace/blob/master/main.py#L81), and it's being iterated on methods from different threads [here](https://github.com/kytos-ng/sdntrace/blob/master/tracing/tracer.py#L151) and [here](https://github.com/kytos-ng/sdntrace/blob/master/tracing/tracer.py#L173), could you double check if I followed correctly?

Reply:
Thank you for looking into this, @viniarck! I investigated this variable in my first analysis, but I didn't change it because I didn't see it as a potential vulnerability to the RuntimeError we discussed. Please correct me if you see another thing. The test I did to validate this was:

```
> python3
Python 3.9.7 (default, Sep  3 2021, 12:37:55)
[Clang 12.0.5 (clang-1205.0.22.9)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> l = list(range(10))
>>> for I in l:
...     if i == 8:
...             l.append(99)
...     print(i)
...
0
1
2
3
4
5
6
7
8
9
99
>>>
```

However, I saw another potential problem with [removing items while interacting](https://github.com/kytos-ng/sdntrace/blob/master/tracing/tracer.py#L176).

Reply:
> 
> However, I saw another potential problem with [removing items while interacting](https://github.com/kytos-ng/sdntrace/blob/master/tracing/tracer.py#L176).

Commit 39d81a7

Reply:
Indeed @italovalcy, you're right. Appending, removing or popping elements from a list doesn't result in a similar or equivalent issue. Let me mark this conversation as resolved.